src/collapse/collapse.js

```diff
@@ -11,28 +11,31 @@ angular.module('ui.bootstrap.collapse', [])
         }
 
         function expand() {
-          element.removeClass('collapse')
-            .addClass('collapsing')
-            .attr('aria-expanded', true)
-            .attr('aria-hidden', false);
+          $q.when(scope.$eval(attrs.expanding), function() {
```

Comment:
In addition, it would be better to cache the expression via `var expandingExpr = $parse(attrs.expanding)` earlier (and similarly for the other expressions), and then one can execute the expressions by doing `expandingExpr(scope)`.